src/meta/src/manager/catalog/mod.rs

```diff
@@ -2193,6 +2194,47 @@ impl CatalogManager {
         Ok(())
     }
 
+    pub async fn comment_on(&self, comment: Comment) -> MetaResult<NotificationVersion> {
+        let core = &mut *self.core.lock().await;
+        let database_core = &mut core.database;
+        database_core.ensure_table_view_or_source_id(&comment.table_id)?;
+
+        let mut tables = BTreeMapTransaction::new(&mut database_core.tables);
+
+        let mut table = tables
+            .get_mut(comment.table_id)
+            .ok_or_else(|| anyhow!("table id {} not found", comment.table_id))?;
```

Comment:
We can directly unwrap here if you changed L2200, otherwise use `MetaError::catalog_id_not_found` instead.